example_notebooks/Process_fake_image_data.ipynb

```diff
@@ -0,0 +1,346 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from matplotlib import pyplot as plt\n",
+    "from rixs import process2d\n",
+    "\n",
+    "%matplotlib widget"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib widget"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Make fake data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fake_curvature = dict(x0=1000, x1=0.04, x2=0.00001)\n",
+    "photon_events = process2d.make_fake_image(**fake_curvature)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Display data in scatterplot"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/mdean/my-envs/mark_six/lib/python3.6/site-packages/matplotlib/__init__.py:942: MatplotlibDeprecationWarning: nbagg.transparent is deprecated and ignored. Use figure.facecolor instead.\n",
```

Comment:
@mpmdean o make it easier to find here is an example of a file path in the notebook source code (it may not appear obvious when viewing it in jupyter-hub that this is here and appears to relate to text in an error/warning message that comes up after running a cell). There are others at lines 111, 221 and 277. Perhaps we just need to make sure that the notebook cells have not yet been run before saving it.

Reply:
This is a tricky issue that we don't have a firm recommendation on yet. Committing notebooks is definitely OK, but committing notebooks _with the output areas_ (i.e. notebooks that have been run) has trade-offs. The advantage of including output areas is that users can then see what the results looks like when they view the notebook on GitHub. One disadvantage is that the outputs can contain user-specific details (the output areas that @awalter-bnl flagged). Another is that it can become quite large in terms of bytes, which is problematic for reasons [documented here](https://nsls-ii.github.io/scientific-python-cookiecutter/including-data-files.html#consider-alternatives).

There is a technical best-of-both worlds solution, but we haven't worked out the details to make it slick and easy to use yet.

I am inclined to let this go as is for now and clean it up once we have that solution. Meanwhile, I encourage restraint -- just don't add too many of these notebooks. :- )